syncs/syncs.go

```diff
@@ -152,3 +153,67 @@ func (s Semaphore) TryAcquire() bool {
 func (s Semaphore) Release() {
 	<-s.c
 }
+
+// Map is a Go map protected by a sync.RWMutex.
```

Comment:
I think this deserves a comment saying that it should not be copied.

Reply:
Done. Since this includes a `sync.RWMutex`,  `go vet` will [complain if this type is shallow copied](https://golang.org/issues/8005).